### ISSUE TITLE
Add Safari support to web extension (desktop and mobile)

### ISSUE THREAD

**Issue by markmhendrickson** [leather-io/extension#706]
The web extension currently supports installation for only Chrome and Firefox browsers. Safari should be supported next given its general popularity.

Blocked by https://github.com/hirosystems/stacks-wallet-web/issues/706

**Comment by Zk2u** [comment]
Completely agree. I use Safari as my main browser since IMO it's the one that matches my tech stack best. It's annoying having to switch between Brave & Safari when testing my apps. This is essential.

**Comment by markmhendrickson** [comment]
In the meantime we could also handle Safari better. Right now we send users to the install page regardless whereas we could simply tell them here to switch to a supported browser:

![image](https://user-images.githubusercontent.com/28991/117654725-715c0980-b196-11eb-9a8b-a4b9c099578a.png)

**Comment by Zk2u** [comment]
Until you guys do this officially, I'm gonna be compiling versions as you release them for Safari [here](https://github.com/syvita/safari-stacks-wallet).

**Comment by markmhendrickson** [comment]
Very cool. We could include this as an unofficial option on https://hiro.so/wallet/install-web in the meantime. Mind providing instructions for installing that we can add there?

**Comment by Zk2u** [comment]
Let me test it out to make sure it works okay, then I'll get back to you :)

**Comment by andresgalante** [comment]
Let's run a regression test in the Safari version to make sure it works. I recall that when we tried this there was a critical feature that didn't work, but I can't put my finger on what it was

**Comment by Zk2u** [comment]
<img width="461" alt="Screenshot 2021-05-10 at 13 53 05" src="https://user-images.githubusercontent.com/67651251/117662062-0d861080-b197-11eb-9a59-d680780159f3.png">
hahaha yeah that might be a slight issue

**Comment by Zk2u** [comment]
When converting it shows this error.
```
Warning: The following keys in your manifest.json are not supported by your current version of Safari. If these are critical to your extension, you should review your code to see if you need to make changes to support Safari:
        manifest_version
        description
        persistent
        version
        js
        content_security_policy
        matches
        commands
        options_ui
        activeTab
        scripts
        browser_action
        web_accessible_resources
        name
        icons
   ```

**Comment by Zk2u** [comment]
The issue appears to be the main content isn't rehydrating

**Comment by aulneau** [comment]
Yeah part of why we have not dug into this yet is it's likely going to take a few cycles to get correct, is not a quick fix :) but please feel free to explore and dig into it if you want, as I said before prs welcome, but safari support isn't our top priority now. It would be _awesome_ to get some contributions from the community for tasks like these 👍

**Comment by Zk2u** [comment]
please can we get this moved up in the priority list? apple just added support for extensions in safari on iPadOS and iOS too. i think it's vital we get this out ASAP, because i really want to be able use my wallet on iOS. should be sorta simple to make the extension an app too. idk, but please let's just get this shipped 🔥

**Comment by markmhendrickson** [comment]
@SyAsteria We can take a look at making this a priority for Q4. If you need this earlier, we'd love to review a PR!

**Comment by markmhendrickson** [comment]
An added benefit here would be supporting mobile (iOS) in addition to desktop out of the gate: https://developer.apple.com/videos/play/wwdc2021/10104/

**Comment by larrysalibra** [comment]
Like @0xAsteria, Safari is also my go to browser, so I was just trying this out today on Safari 15 and Xcode 13.

<img width="905" alt="Screen Shot 2021-09-22 at 10 08 48 PM" src="https://user-images.githubusercontent.com/597182/134359723-107f7118-4f1e-4266-840a-89eee772744e.png">
<img width="976" alt="Screen Shot 2021-09-22 at 10 12 17 PM" src="https://user-images.githubusercontent.com/597182/134360055-114f62a0-33c8-4f5d-b496-b0911389b2f1.png">

Good news is the "Warning: The following keys in your manifest.json are not supported by your current version of Safari. If these are critical to your extension, you should review your code to see if you need to make changes to support Safari:" errors are gone!

More good news is that it responds to a the connect sign in event on btc.us.

Bad news is that it looks like this:


<img width="1435" alt="Screen Shot 2021-09-22 at 10 16 22 PM" src="https://user-images.githubusercontent.com/597182/134360818-2dbd4924-d5db-4b86-a0b0-4e19a9b0abe1.png">

The error in the extension is "unable to get origin from request"

<img width="1258" alt="Screen Shot 2021-09-22 at 10 16 35 PM" src="https://user-images.githubusercontent.com/597182/134361109-3a923cdf-ff60-45a1-85f0-df379be2285e.png">

**Comment by kyranjamie** [comment]
Thanks @larrysalibra, interesting that the TypeError is thrown, as its inside a try/catch

**Comment by markmhendrickson** [comment]
@larrysalibra super helpful, thanks! 

Is our sense here that getting the extension supported for Safari might be a relatively modest compatibility tweak undertaking?

**Comment by kyranjamie** [comment]
I think Safari desktop should be quite manageable. But it's not 1:1 with mobile. Non-persistent background pages for example may cause some issues.

**Comment by nargetdev** [comment]
This is predominately necessary for enabling native application development on iOS via ionicframework.  I tried going the cocoapods route too but [that repository](https://github.com/stacks-network/blockstack-ios) is 2 years old and unmaintained (broken).
